### PR TITLE
feat: add uptime alerts commands

### PR DESCRIPTION
**Pull request by mikesmithgh**
PR for https://github.com/digitalocean/doctl/issues/1419 as part of #hacktoberfest

Checklist:
- [x] merge https://github.com/digitalocean/godo/pull/637 and vendor project
- [x] write tests

### CODE REVIEW

**Comment by mikesmithgh** [comment]
Hi @danaelhe this PR is ready for review. I am leaving it in draft until its dependent PR https://github.com/digitalocean/godo/pull/637 is merged. After that is merged, I'll run the vendor command and update this PR.

It looks like I made a lot of changes because I ran `make mocks` to generate the mocks for `uptime_alerts_test.go`. A new version of gomock was released recently which changes the signature on all the mocks causing all those updates.

Please let me know what you think. Thanks!

**Comment by danaelhe** [comment]
> Hi @danaelhe this PR is ready for review. I am leaving it in draft until its dependent PR [digitalocean/godo#637](https://github.com/digitalocean/godo/pull/637) is merged. After that is merged, I'll run the vendor command and update this PR.
> 
> It looks like I made a lot of changes because I ran `make mocks` to generate the mocks for `uptime_alerts_test.go`. A new version of gomock was released recently which changes the signature on all the mocks causing all those updates.
> 
> Please let me know what you think. Thanks!

Awesome! I'll cut a new release of godo as well and then review this PR. That's totally fine. Thank you for the update!

**Comment by mikesmithgh** [comment]
> Awesome! I'll cut a new release of godo as well and then review this PR. That's totally fine. Thank you for the update!

@danaelhe Thanks! I just merged the changes and it ready for review 🙂

File: commands/uptime_alerts.go [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+	}
+
+	cmdUptimeAlertsCreate := CmdBuilder(cmd, RunUptimeAlertsCreate, "create <uptime-check-id>", "Create an uptime alert", `Use this command to create an uptime alert on your account.
+
+You can use flags to specify the uptime alert, type, threshold, comparison, notifications, and period.`, Writer,
+		aliasOpt("c"), displayerType(&displayers.UptimeAlert{}))
+	AddStringFlag(cmdUptimeAlertsCreate, doctl.ArgUptimeAlertName, "", "", "Uptime alert name", requiredOpt())
+	AddStringFlag(cmdUptimeAlertsCreate, doctl.ArgUptimeAlertType, "", "", "Uptime alert type, must be one of latency, down, down_global or ssl_expiry", requiredOpt())
+	AddIntFlag(cmdUptimeAlertsCreate, doctl.ArgUptimeAlertThreshold, "", 0, "Uptime alert threshold at which the alert will enter a trigger state. The specific threshold is dependent on the alert type.")
+	AddStringFlag(cmdUptimeAlertsCreate, doctl.ArgUptimeAlertComparison, "", "", "Uptime alert comparison operator used against the alert's threshold.")
```
Comment (danaelhe) [inline commands/uptime_alerts.go:48]:
```suggestion
	AddStringFlag(cmdUptimeAlertsCreate, doctl.ArgUptimeAlertComparison, "", "", "Uptime alert comparison operator used against the alert's threshold. Must be either `greater_than` or `less_than`")
```

File: commands/uptime_alerts.go [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+	AddStringSliceFlag(cmdUptimeAlertsCreate, doctl.ArgUptimeAlertEmails, "", nil, "Emails to send uptime alerts to.")
```
Comment (danaelhe) [inline commands/uptime_alerts.go:49]:
```suggestion
	AddStringSliceFlag(cmdUptimeAlertsCreate, doctl.ArgUptimeAlertEmails, "", nil, "Emails to send uptime alerts to. Must be emails associated with the DigitalOcean account")
```

File: commands/uptime_alerts.go [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+		aliasOpt("ls"), displayerType(&displayers.UptimeAlert{}))
+
+	cmdUptimeAlertsUpdate := CmdBuilder(cmd, RunUptimeAlertsUpdate, "update <uptime-check-id> <uptime-alert-id>", "Update an uptime alert", `Use this command to update an uptime alert on your account.
+
+You can use flags to specify the uptime alert, type, threshold, comparison, notifications, and period.`, Writer,
+		aliasOpt("u"), displayerType(&displayers.UptimeAlert{}))
+	AddStringFlag(cmdUptimeAlertsUpdate, doctl.ArgUptimeAlertName, "", "", "Uptime alert name", requiredOpt())
+	AddStringFlag(cmdUptimeAlertsUpdate, doctl.ArgUptimeAlertType, "", "", "Uptime alert type, must be one of latency, down, down_global or ssl_expiry", requiredOpt())
+	AddIntFlag(cmdUptimeAlertsUpdate, doctl.ArgUptimeAlertThreshold, "", 0, "Uptime alert threshold at which the alert will enter a trigger state. The specific threshold is dependent on the alert type.")
+	AddStringFlag(cmdUptimeAlertsUpdate, doctl.ArgUptimeAlertComparison, "", "", "Uptime alert comparison operator used against the alert's threshold.")
```
Comment (danaelhe) [inline commands/uptime_alerts.go:67]:
```suggestion
	AddStringFlag(cmdUptimeAlertsUpdate, doctl.ArgUptimeAlertComparison, "", "", "Uptime alert comparison operator used against the alert's threshold. Must be either `greater_than` or `less_than`")
```

File: commands/uptime_alerts.go [hunk 4]
```diff
@@ -0,0 +1,347 @@
...
+	AddStringSliceFlag(cmdUptimeAlertsUpdate, doctl.ArgUptimeAlertEmails, "", nil, "Emails to send uptime alerts to.")
```
Comment (danaelhe) [inline commands/uptime_alerts.go:68]:
```suggestion
	AddStringSliceFlag(cmdUptimeAlertsUpdate, doctl.ArgUptimeAlertEmails, "", nil, "Emails to send uptime alerts to. Must be emails associated with the DigitalOcean account")
```

File: do/uptime_checks.go [hunk 5]
```diff
@@ -24,19 +24,29 @@ type UptimeCheck struct {
...
+	*godo.UptimeAlert
+}
+
 // UptimeCheckState is a wrapper for godo.UptimeCheckState.
 type UptimeCheckState struct {
 	*godo.UptimeCheckState
 }
 
-// UptimeChecksService is an interface for interacting with DigitalOcean's volume api.
+// UptimeChecksService is an interface for interacting with DigitalOcean's uptime check api.
```
Comment (danaelhe) [inline do/uptime_checks.go:37]: good catch 👍

**Review by danaelhe — DISMISSED** [review]
Look fantastic! This is an extremely high valuable contribution- thank you so much! I just had a few nit-picks on the documentation but other than that it's ready to go 🚀

**Comment by mikesmithgh** [comment]
> Look fantastic! This is an extremely high valuable contribution- thank you so much! I just had a few nit-picks on the documentation but other than that it's ready to go 🚀

Thanks! I'll make the updates and let you know

**Comment by mikesmithgh** [comment]
> > Look fantastic! This is an extremely high valuable contribution- thank you so much! I just had a few nit-picks on the documentation but other than that it's ready to go 🚀
> 
> Thanks! I'll make the updates and let you know

@danaelhe done!

**Review by danaelhe — APPROVED** [review]
🚀

**Comment by mikesmithgh** [comment]
> 🚀

🎉 awesome, thanks!

**Pull request merged**